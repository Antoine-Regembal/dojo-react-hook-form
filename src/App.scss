.grid {
  display: grid;
  width: 100%;
  grid-template-columns: 2fr 1fr;
}

.area {
  height: 100vh;
  overflow-y: scroll;

  &--documentation {
    background-color: white;
    padding: 0 1rem 0 1rem;
  }
  &--form {
    background-color: whitesmoke;
  }
}

.form {
  padding: 3rem 0 0 3rem;

  &__label {
    display: block;

    &--required::after {
      content: "*";
      color: red;
    }
  }
  &__input {
    display: block;
    margin: 0.3em 0 0.3em 0;
    padding: 0.3rem 0.8rem 0.3rem 0.8rem;

    &--error {
      color: red;
      border-color: red;
      border-style: solid;
    }
  }
  &__button {
    margin-top: 3rem;
    padding: 0.3rem 0.8rem 0.3rem 0.8rem;
  }
  &__error {
    display: block;
    color: red;
  }
  &--required::before {
    content: "* required fields";
    color: red;
    font-size: small;
  }
}

.input-wrapper:not(:first-of-type) {
  margin-top: 2em;
}

em {
  font-size: small;
}
