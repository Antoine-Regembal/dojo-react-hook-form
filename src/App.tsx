import "./App.scss";

export const App = () => (
	<div>
		<div></div>
		<div></div>
	</div>
);