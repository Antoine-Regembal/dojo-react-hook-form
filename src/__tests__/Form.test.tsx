import {render} from "@testing-library/react";
import { App } from "../App";

describe("Form", function () {
	describe("snapshots", function () {
		it("should match empty form", function () {
			const {container} = render(<App />);
			expect(container).toMatchSnapshot();
		});
	});
});
