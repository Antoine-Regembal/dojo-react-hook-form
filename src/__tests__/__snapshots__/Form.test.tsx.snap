// Vitest Snapshot v1

exports[`Form > snapshots > should match empty form 1`] = `
<div>
  <div
    class="grid"
  >
    <div
      class="area area--documentation"
    >
      <div>
        <header>
          <h1>
            DOJO : react-hook-form
          </h1>
        </header>
        <nav>
          <ul
            class="nav"
          >
            <a
              class="nav__list"
              href="/prerequisites"
            >
              <button
                class="nav__list__button nav__list__button--selected"
                type="button"
              >
                prerequisites
              </button>
            </a>
            <a
              class="nav__list"
              href="/subject"
            >
              <button
                class="nav__list__button"
                type="button"
              >
                subject
              </button>
            </a>
          </ul>
        </nav>
        <section>
          <h2
            class="section__title--before-starting"
          >
            Prerequisites before starting the exercises
          </h2>
          <section>
            <h3>
              Required packages version to run app and tests
            </h3>
            <ul>
              <li>
                <a
                  href="https://nodejs.org/en/"
                  rel="norefferer noreferrer"
                  target="_blank"
                >
                  node at version 18.0.0 or more
                </a>
              </li>
            </ul>
          </section>
          <section>
            <h3>
              Explore a little bit the react-hook-form documentation
            </h3>
            <p>
              To understand a little more on what this DOJO is about, explore a little bit
               
              <a
                href="https://react-hook-form.com"
                rel="noreferrer"
                target="_blank"
              >
                the official react-hook-form
              </a>
               
              documentation
            </p>
          </section>
          <section>
            <h3>
              Find out which packages are used in this projects
            </h3>
            <section>
              <h4>
                To run this react application
              </h4>
              <ul>
                <li>
                  <a
                    href="https://vitejs.dev"
                    rel="noreferrer"
                    target="_blank"
                  >
                    vite
                  </a>
                </li>
              </ul>
            </section>
            <section>
              <h4>
                For coding
              </h4>
              <ul>
                <li>
                  <a
                    href="https://reactjs.org"
                    rel="noreferrer"
                    target="_blank"
                  >
                    react
                  </a>
                </li>
                <li>
                  <a
                    href="https://www.typescriptlang.org"
                    rel="noreferrer"
                    target="_blank"
                  >
                    typescript
                  </a>
                </li>
                <li>
                  <a
                    href="https://sass-lang.com"
                    rel="noreferrer"
                    target="_blank"
                  >
                    saas
                  </a>
                </li>
              </ul>
            </section>
            <section>
              <h4>
                For testing
              </h4>
              <ul>
                <li>
                  <a
                    href="https://vitest.dev"
                    rel="noreferrer"
                    target="_blank"
                  >
                    vitest
                  </a>
                </li>
                <li>
                  <a
                    href="https://www.npmjs.com/package/specflow-emulator"
                    rel="noreferrer"
                    target="_blank"
                  >
                    specflow-emulator
                  </a>
                </li>
                <li>
                  <a
                    href="https://cucumber.io/docs/gherkin/"
                    rel="noreferrer"
                    target="_blank"
                  >
                    cucumber gherkin
                  </a>
                </li>
              </ul>
            </section>
          </section>
        </section>
        <section>
          <h2
            class="section__title--install"
          >
            Install the React Developer Tool
          </h2>
          <ul>
            <li>
              <a
                href="https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi"
                rel="noreferrer"
                target="_blank"
              >
                for Chrome
              </a>
            </li>
            <li>
              <a
                href="https://addons.mozilla.org/fr/firefox/addon/react-devtools/"
                rel="noreferrer"
                target="_blank"
              >
                for Firefox
              </a>
            </li>
            <li>
              <a
                href="https://stackoverflow.com/questions/65373114/i-want-to-add-react-devtools-to-safari-browser"
                rel="noreferrer"
                target="_blank"
              >
                for Safari
              </a>
            </li>
          </ul>
        </section>
      </div>
    </div>
    <div
      class="area area--form"
    >
      <form
        class="form form--required"
      >
        <div
          class="input-wrapper"
        >
          <label
            class="form__label form__label--required"
            for="firstname"
          >
            Firstname
          </label>
          <input
            class="form__input"
            id="firstname"
            placeholder="Firstname"
            type="text"
            value=""
          />
          <em
            class="form__error"
            role="alert"
          />
        </div>
        <div
          class="input-wrapper"
        >
          <label
            class="form__label form__label--required"
            for="lastname"
          >
            Lastname
          </label>
          <input
            class="form__input"
            id="lastname"
            placeholder="Lastname"
            type="text"
            value=""
          />
          <em
            class="form__error"
            role="alert"
          />
        </div>
        <div
          class="input-wrapper"
        >
          <label
            class="form__label form__label--required"
            for="email"
          >
            Email
          </label>
          <input
            class="form__input"
            id="email"
            placeholder="Email"
            type="text"
            value=""
          />
          <em
            class="form__error"
            role="alert"
          />
        </div>
        <div
          class="input-wrapper"
        >
          <label
            class="form__label form__label--required"
            for="confirmEmail"
          >
            Confirm email
          </label>
          <input
            class="form__input"
            id="confirmEmail"
            placeholder="Confirm email"
            type="text"
            value=""
          />
          <em
            class="form__error"
            role="alert"
          />
        </div>
        <button
          class="form__button"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
